0001-kaa-protocol/README.md

```diff
@@ -6,35 +6,83 @@ editor: Alexey Shmalko <ashmalko@cybervisiontech.com>
+  - [Security](#security)
+- [Open questions](#open-questions)
+  - [Topic name aliases](#topic-name-aliases)
+  - [Status topics](#status-topics)
+  - [Endpoint migration](#endpoint-migration)
+    - [UC1](#uc1)
+    - [UC2](#uc2)
+  - [Reporting errors](#reporting-errors)
+  - [Security](#security)
+
 ## Introduction
-The document describes the general requirements, principles, design and guidelines for the **Kaa Protocol** (KP) version 1.
-KP is a standard protocol designed for connecting clients and endpoints (colloquially - devices) to a Kaa server.
+This document describes general requirements, principles, design, and guidelines for **Kaa Protocol** (KP) version 1.
+KP is a standard protocol designed to connect client applications and endpoints (colloquially --- devices) to a Kaa server.
 
 ## Language
-
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
 
+The following terms and definitions are used in this RFC.
+
+- **Endpoint**: an entity managed by Kaa platform.
+Platform users are interested in differentiating all endpoints.
+Endpoints may be either physical or virtual.
+
+- **Client**: an application that uses a single "connection" to the server.
+One client may represent multiple endpoints.
+
+- **Endpoint token**: an opaque data blob that uniquely identifies an endpoint.
+Each endpoint has exactly one token assigned by the server.
+
+- **Resource path**: a unique resource identifier included in each request.
+For MQTT, that is Topic Name, for CoAP --- URI-Paths.
+
+- **Extension**: a piece of functionality offered to the client by the server.
+Extension is usually represented by a separate resource.
+Examples of extensions are [Data Collection]()<!--TODO-->, [Configuration Management]()<!--TODO-->, [Endpoint Metadata Synchronization]()<!--TODO-->.
+
 ## Requirements and constraints
-- KP must support asynchronous messaging.
-- Clients must be able to connect to a server without using an SDK and with minimum custom implementation.
-- Encrypted and unencrypted channels must be supported.
-- KP must be able to work over MQTT v3.1 and later.
-- KP should be able to work over CoAP.
-- KP must be extensible to support future extensions.
-- Clients should be able to work through MQTT gateways. That also includes MQTT-SN gateways.
-- KP must support client authentication.
-- KP must support endpoint identification (but not necessarily authentication).
+
+KP requirements:
+
+- Asynchronous messaging.
+- Encrypted and unencrypted channels.
+- Working over [MQTT](http://mqtt.org/) version 3.1 and newer.
+- Working over [CoAP](http://coap.technology/).
+- Extensibility, support for future extensions.
+- Client authentication.
+- Endpoint identification (but not necessarily authentication).
+
+Client requirements:
```

Comment:
These are requirements to the protocol, not client. i.e., `the protocol should allow client to ...`